### ISSUE TITLE
[FE][TECH DEBT] - Refactor and reduce HOC useage

### ISSUE THREAD

**Issue by brianseek** [department-of-veterans-affairs/va.gov-team#49081]
## Description
We have 7 HOCs in the `containers` directory. I think one isn't in-use. We should evaluate if all are needed and if we can consolidate the usage a bit. The data persistence work also adds a component based wrapper where some of this functionality could reside.

## Tasks
- [ ] [FE] evaluate HOCs
- [ ] [FE] refactor HOCs
- [ ] Create or update e2e tests
- [ ] Update relevant documentations (readme and use cases)

## Acceptance Criteria
- [ ] Passes Accessibility checks using axe
- [ ] Deployed to staging.
- [ ] Tested on staging
- [ ] Designer has signed off on the feature
- [ ] Product team has signed off on the  feature
- [ ] Prod validation:
  - [ ] not necessary
  - [ ] scheduled

## Code Review Steps
- [ ] Read ticket details and verify that ACs are met
- [ ] Review code in PR
- [ ] Run branch locally
- [ ] Check that new and updated translations use new keys
- [ ] Check for hardcoded strings

**Comment by brianseek** [comment]
After trying a few ideas and not really coming up with a better solution. I landed on an opinion that the way we have things is the most readable and not necessarily bad. Some of the options I was looking at added tech debt in different places. Also took some time to read up on if we would have any performance gains moving away from HOCs and didn't really see anything. In one case I saw a person making an argument that HOCs are more performant than custom hooks. 

Tagging @acrollet & @wullaski for validation on this one before closing out.
